### PR TITLE
[PDI-14734]-Dialog's table discards input after losing focus

### PR DESCRIPTION
**Pull request by denisprotopopov**
- fixed saving dialog

### CODE REVIEW

**Comment by denisprotopopov** [comment]
@mbatchelor please reveiw the changes

File: ui/src/org/pentaho/di/ui/job/entries/copyfiles/JobEntryCopyFilesDialog.java [hunk 1]
```diff
@@ -527,28 +527,31 @@ public void getData() {
...
-          String clusterName = jobEntry.getConfigurationBy( sourceUrl );
-          ti.setText( 1, STATIC_ENVIRONMENT );
-          if ( clusterName != null ) {
-            clusterName =
-                clusterName.startsWith( JobEntryCopyFiles.LOCAL_SOURCE_FILE ) ? LOCAL_ENVIRONMENT : clusterName;
-            clusterName =
-                clusterName.startsWith( JobEntryCopyFiles.STATIC_SOURCE_FILE ) ? STATIC_ENVIRONMENT : clusterName;
-
-            ti.setText( 1, clusterName );
+          String clusterName1 = jobEntry.getConfigurationBy( sourceUrl );
```
Comment (mbatchelor) [inline ui/src/org/pentaho/di/ui/job/entries/copyfiles/JobEntryCopyFilesDialog.java]: Why change the variable name to clusterName1?

File: ui/src/org/pentaho/di/ui/job/entries/copyfiles/JobEntryCopyFilesDialog.java [hunk 2]
```diff
@@ -527,28 +527,31 @@ public void getData() {
...
+          } else {
+            sourceUrl = "";
+          }
           ti.setText( 2, sourceUrl );
         }
         if ( jobEntry.wildcard[i] != null ) {
           ti.setText( 3, jobEntry.wildcard[i] );
         }
-        if ( jobEntry.destination_filefolder[i] != null && !Utils.isEmpty( jobEntry.destination_filefolder[i] ) ) {
+        if ( jobEntry.destination_filefolder[i] != null ) {
```
Comment (mbatchelor) [inline ui/src/org/pentaho/di/ui/job/entries/copyfiles/JobEntryCopyFilesDialog.java]: Why this change? Are you suddenly allowing empty destination folders ?

File: engine/test-src/org/pentaho/di/job/entries/copyfiles/JobEntryCopyFilesTest.java [hunk 3]
```diff
@@ -74,6 +74,7 @@ public void fileCopied() throws Exception {
 
     verify( entry ).processFileFolder( anyString(), anyString(),
         anyString(), any( Job.class ), any( Result.class ) );
+    verify( entry, atLeast( 1 ) ).preprocessfilefilder( any( String[].class ) );
```
Comment (mbatchelor) [inline engine/test-src/org/pentaho/di/job/entries/copyfiles/JobEntryCopyFilesTest.java]: I don't see how this one test covers the fact that you are now persisting and restoring job entries with  "EMPTY_DEST_URL-" and "EMPTY_SOURCE_URL-" prefixes.

**Comment by denisprotopopov** [comment]
sent a new PR https://github.com/pentaho/pentaho-kettle/pull/3926
